File: services/semantic/src/domain/models/paper.py
class PaperModel:
    def __init__(
        self,
        ID,
        Title,
        Abstract,
        Year,
        Best_oa_location,
        Referenced_works=None,
        Related_works=None,
        Cited_by_count=0,
        Authors=None,
        Institutions=None,
        Identifiers=None,
    ):
        self.ID = ID
        self.Title = Title
        self.Abstract = Abstract
        self.Year = Year
        self.Best_oa_location = Best_oa_location
        self.Referenced_works = list(Referenced_works or [])
        self.Related_works = list(Related_works or [])
        self.Cited_by_count = Cited_by_count
        self.Authors = list(Authors or [])
        self.Institutions = list(Institutions or [])
        self.Identifiers = list(Identifiers or [])

File: services/gateway/internal/transport/http/presenters/paper.go
package presenters

type AddPaperRequest struct {
	Id               string            `json:"id"`
	Title            string            `json:"title"`
	Abstract         string            `json:"abstract"`
	Year             int               `json:"year"`
	Best_oa_location string            `json:"best_oa_location"`
	ReferencedPapers []ReferencedPaper `json:"referenced_paper"`
	RelatedPaper     []RelatedPaper    `json:"related_paper"`
}

type Paper struct {
	Id               string            `json:"id"`
	Title            string            `json:"title"`
	Abstract         string            `json:"abstract"`
	Year             int               `json:"year"`
	Best_oa_location string            `json:"best_oa_location"`
	State            string            `json:"state"`
	ReferencedWorks  []string          `json:"referenced_works"`
	RelatedWorks     []string          `json:"related_works"`
	CitedByCount     int               `json:"cited_by_count"`
	Authors          []string          `json:"authors"`
	Institutions     []string          `json:"institutions"`
	Identifiers      []PaperIdentifier `json:"identifiers"`
}

type PaperIdentifier struct {
	Type  string `json:"type"`
	Value string `json:"value"`
}

type ReferencedPaper struct {
	Id string `json:"id"`
}

type RelatedPaper struct {
	Id string `json:"id"`
}
